packages/governance/test/swingsetTests/contractGovernor/bootstrap.js

```diff
@@ -198,6 +198,31 @@ const setupParameterChanges = async (
   return { details, outcomeOfUpdate };
 };
 
+const setupOfferFilterChange = async (
+  zoe,
+  log,
+  governor,
+  installations,
+  invitation,
+  strings = ['foo', 'bar:'],
+) => {
+  const filterChangeSpec = harden(strings);
+  /** @type {ContractGovernanceVoteResult} */
+  const { details, instance, outcomeOfUpdate } = await E(
+    governor,
+  ).voteOnOfferFilter(installations.binaryVoteCounter, 2n, filterChangeSpec);
+
+  E(E(zoe).getPublicFacet(instance))
+    .getOutcome()
+    .then(outcome => log(`vote outcome: ${q(outcome)}`))
+    .catch(e => log(`vote failed ${e}`));
+
+  E.when(outcomeOfUpdate, outcome => log(`updated to (${q(outcome)})`)).catch(
```

Comment:
TIL `E.when`  branches off the deferred resolution without affecting other consumers of `outcomeOfUpdate` resolution.